nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/DeleteHBaseRow.java

```diff
@@ -0,0 +1,183 @@
+            .name("delete-hb-row-ff-count")
+            .displayName("Batch Size")
+            .description("The number of deletes to send per batch.")
+            .required(true)
+            .defaultValue("50")
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .expressionLanguageSupported(false)
+            .build();
+
+    static final PropertyDescriptor KEY_SEPARATOR = new PropertyDescriptor.Builder()
+            .name("delete-hb-separator")
+            .displayName("Delete Row Key Separator")
+            .description("The separator character(s) that separate multiple row keys " +
+                    "when multiple row keys are provided in the flowfile body")
+            .required(true)
+            .defaultValue(",")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+    static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+            .name("Character Set")
+            .description("The character set used to encode the row key for HBase.")
+            .required(true)
+            .defaultValue("UTF-8")
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = super.getSupportedPropertyDescriptors();
+        properties.add(ROW_ID_LOCATION);
+        properties.add(FLOWFILE_FETCH_COUNT);
+        properties.add(BATCH_SIZE);
+        properties.add(KEY_SEPARATOR);
+        properties.add(CHARSET);
+
+        return properties;
+    }
+
+    @Override
+    protected void doDelete(ProcessContext context, ProcessSession session) throws Exception {
+        final int batchSize      = context.getProperty(BATCH_SIZE).asInteger();
+        final String location    = context.getProperty(ROW_ID_LOCATION).getValue();
+        final int flowFileCount  = context.getProperty(FLOWFILE_FETCH_COUNT).asInteger();
+        final String charset     = context.getProperty(CHARSET).getValue();
+        List<FlowFile> flowFiles = session.get(flowFileCount);
+
+        if (flowFiles != null && flowFiles.size() > 0) {
+            for (int index = 0; index < flowFiles.size(); index++) {
+                FlowFile flowFile = flowFiles.get(index);
+                try {
+                    if (location.equals(ROW_ID_BODY.getValue())) {
+                        flowFile = doDeleteFromBody(flowFile, context, session, batchSize, charset);
+                        Relationship rel = flowFile.getAttribute("restart.index") != null ? REL_FAILURE : REL_SUCCESS;
+                        session.transfer(flowFile, rel);
+                    } else {
+                        doDeleteFromAttribute(flowFile, context, charset);
+                        session.transfer(flowFile, REL_SUCCESS);
+                    }
```

Comment:
It's possible that I overlook but do we send provenance events when this processor deletes rows from HBase? If not, please add that around here. Provenance events `transit url` is important for things like NiFi Atlas integration. [AbstractPutHBase](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-hbase-bundle/nifi-hbase-processors/src/main/java/org/apache/nifi/hbase/AbstractPutHBase.java#L198) can be a reference, but since it's delete objects, call [ProvenanceReporter.invokeRemoteProcess](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/StandardProvenanceReporter.java#L238) instead.

Reply:
Done